### PR TITLE
Mk22fx512

### PR DESCRIPTION
**Pull request by ostaebler**
Add support for MK22FX512 (mk22fa12).

I'm not sure what to write in mcux/README. I don't know how I can find the 'Tag'.

### CODE REVIEW

**Comment by ostaebler** [comment]
Please see the corresponding PR for zephyr

https://github.com/zephyrproject-rtos/zephyr/pull/18933

File: mcux/CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +37,9 @@ if (${MCUX_DEVICE} MATCHES "LPC")
   zephyr_sources(devices/${MCUX_DEVICE}/fsl_power.c)
   zephyr_sources(devices/${MCUX_DEVICE}/fsl_reset.c)
 endif()
+if (${MCUX_SERIES} MATCHES "K2X")
+  zephyr_sources(devices/${MCUX_DEVICE}/system_${MCUX_DEVICE}.c)
```
Comment (bfrog) [inline mcux/CMakeLists.txt]: This breaks the convention of having the setup code of the SOC in zephyr itself rather than in the HAL

Comment (ostaebler) [reply mcux/CMakeLists.txt]: How 'established' is this convention?

E.g. nordic does this the same, see https://github.com/zephyrproject-rtos/hal_nordic/blob/master/CMakeLists.txt#L23

Comment (galak) [reply mcux/CMakeLists.txt]: I think the point is that all other NXP SoCs do it as @brog mentions and the K22F support should match what other NXP Kinetis SoCs are doing.

File: mcux/README [hunk 2]
```diff
@@ -24,6 +24,7 @@ Status:
    MIMXRT1062	SDK 2.5.0 (2018-12-17)		REL_2.5.0_REL9_RFP_RC3_7_1
    MIMXRT1021	SDK 2.5.0 (2018-12-17)		REL_2.5.0_REL9_RFP_RC3_7_1
    MIMXRT1064	SDK 2.4.0 (2018-10-22)		REL_SDK_2.4.0_RT1064_RFP_INTERNAL
+   MK22FX512	SDK 2.3.1 (2018-04-17)		???
```
Comment (MaureenHelm) [inline mcux/README]: I removed the tag in #8.  Please rebase

File: mcux/devices/MK22FA12/fsl_device_registers.h [hunk 3]
```diff
@@ -0,0 +1,64 @@
+/*
+ * The Clear BSD License
```
Comment (MaureenHelm) [inline mcux/devices/MK22FA12/fsl_device_registers.h]: The Zephyr project decided not to accept this license upstream. We need to see if NXP can release an updated MCUXpresso SDK for this SoC.

**Comment by ostaebler** [comment]
Sorry for my late reply.

Unfortunately, I'll not find the time to update this PR.

**Pull request closed without merging**